### PR TITLE
Fix, add metrics testing to tls_tunnel Au test.

### PR DESCRIPTION
**Pull request by ywkaras**

### CODE REVIEW

**Comment by ywkaras** [comment]
Keeping this a draft because I'm not sure the values of the metrics are correct.

**Comment by ywkaras** [comment]
[approve ci docs]

File: proxy/http/HttpTransact.cc [hunk 1]
```diff
@@ -5723,9 +5723,13 @@ HttpTransact::initialize_state_variables_from_request(State *s, HTTPHdr *obsolet
     s->cache_info.action = CACHE_DO_NO_ACTION;
   }
 
+  bool do_increment_stat = (0 == s->method);
+
   s->method = incoming_request->method_get_wksidx();
 
-  if (s->method == HTTP_WKSIDX_GET) {
+  if (!do_increment_stat) {
```
Comment (maskit) [inline proxy/http/HttpTransact.cc:5730]: Why adding test requires this change? Does this actually fix something?

Comment (ywkaras) [reply proxy/http/HttpTransact.cc:5730]: Yes , it fixes the associated issue.

Comment (maskit) [reply proxy/http/HttpTransact.cc:5730]: Name the PR accordingly. Who expects behavior change from the current title.

File: proxy/http/HttpTransact.cc [hunk 2]
```diff
@@ -5723,9 +5723,13 @@ HttpTransact::initialize_state_variables_from_request(State *s, HTTPHdr *obsolet
     s->cache_info.action = CACHE_DO_NO_ACTION;
   }
 
+  bool do_increment_stat = (0 == s->method);
```
Comment (shinrich) [inline proxy/http/HttpTransact.cc:5726]: So without this check unset methods count towards tunnel.   Won't this remove some legitimate tunnel counts like tr_pass tunnels?  In what cases are we seeing unset methods?

**Comment by shinrich** [comment]
Seems reasonable enough.  But a description of what it means for the initialize method to be called with method set to 0 would be useful.  Otherwise, I'm unsure about what is being fixed here.

File: proxy/http/HttpTransact.cc (hunk 2 above)
Comment (ywkaras) [reply proxy/http/HttpTransact.cc:5738]: OK done.

**Comment by ywkaras** [comment]
> Seems reasonable enough. But a description of what it means for the initialize method to be called with method set to 0 would be useful. Otherwise, I'm unsure about what is being fixed here.

I've added a new commit to address this.

**Comment by ywkaras** [comment]
@shinrich ^^^

**Comment by ezelkow1** [comment]
[approve ci centos]

**Review by SolidWallOfCode — APPROVED** [review]

**Pull request merged**